modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/file_entry_history.jsp

```diff
@@ -76,11 +77,10 @@
 		</li>
 
 	<%
-		index++;
 	}
 	%>
 
-	<c:if test='<%= FeatureFlagManagerUtil.isEnabled("LPS-175915") && (index >= fileVersionsLimit) %>'>
+	<c:if test='<%= FeatureFlagManagerUtil.isEnabled("LPS-175915") && (fileEntry.getFileVersionsCount(status) >= 10) %>'>
```

Comment:
the value 10 should be the var `end`, if you change the variable should change it 2 times and maybe you can forget this.